planning/roadmap.md

```diff
@@ -4,93 +4,350 @@ description: Planned work for the 2024 release of Thunderbird.
-* **and everything else...**
-
-We will always have dedicated resources allocated to tackle crashes or performance regressions caused by any of these components.
+Priority: <span style="background-color: #B3FFB3">P3</span>
+Delivery Quarter: Q2 2026
+Why: Increasing the productivity aspect of Thunderbird.
+High level objectives (non-exhaustive):
+* Store notes in the profile and in Sync.
+* HTML, markdown, plaintext editor.
+* Share notes via Send.
+* Copy the Google approach (create notes for this event).
+* Potential Pro offering (cloud notes with shared access like Etherpad).
+* Potential syncing through IMAP.
+
+## Maintenance Priorities
+
+### Support URLs
+Priority: <span style="background-color: #FF8080">P1</span>
+Delivery Quarter: Q1 2025
+Why: Let's be consistent and not lead users to broken links.
+High level objectives (non-exhaustive):
+* Plan and implement an efficient and standard approach to links within the application
+
+### Better logging/diagnostics
+Priority: <span style="background-color: #FF8080">P1</span>
+Delivery Quarter: Q1 2025
+Why: We often find it hard to capture helpful information from users when issues are reported. To save time and frustration on both sides, it would help if our logging and diagnostics were improved.
+High level objectives (non-exhaustive):
+* Implement Sentry and telemetry to capture data that will help us prioritize response to issues based on frequency.
+* Add or change logging tools to make it easier for the majority of users to provide logs.
+
+### Fluent Migration
+Priority: <span style="background-color: #FFFF99">P2</span>
+Delivery Quarter: Q2 2025
+Why: DTD and properties files will one day be removed.
+High level objectives (non-exhaustive):
+* Assign a small migration effort every 2 weeks to each developer.
+
+### Complete Glean testing and uplift
+Priority: <span style="background-color: #FFFF99">P2</span>
+Delivery Quarter: Q2 2025
+Why: Fully commit to our new telemetry SDK
+High level objectives (non-exhaustive):
+* Complete the project.
+* Validation that data is being captured as expected.
+* Help services consume the data.
+
+### Hardening
+Priority: <span style="background-color: #FFFF99">P2</span>
+Delivery Quarter: Q3 2025
+
+### Fuzzing and Monkey testing
+Priority: <span style="background-color: #FFFF99">P2</span>
+Delivery Quarter: Q3 2025
+Why: Our current testing and QA process doesn'�t catch all issues and our users suffer
+High level objectives (non-exhaustive):
+* Work with Rob to create scripts, VMs, checking tools etc that we can have running regularly to catch edge cases much sooner without users having to experience then report.
+
+### Remove IMAP JS
+Priority: <span style="background-color: #B3FFB3">P3</span>
+Delivery Quarter: Q3 2025
+Why: Old approach of rewriting IMAP in JS. Discarded by the core developers as not a viable approach.
+High level objectives (non-exhaustive):
+* Remove unused code.
+* Remove preference.
+* Close all related open bugzilla bugs.
```

Comment:
This was done already, I guess we can remove it?

Reply:
good point. i've removed it.